util/hostnamer_rdns.go

```diff
@@ -0,0 +1,51 @@
+package util
+
+import (
+	"net"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/patrickmn/go-cache"
+)
+
+// HostNamerRDNS is a component that provides cached hostnames for IP
+// addresses passed as strings, determined via reverse DNS lookups.
+type HostNamerRDNS struct {
+	Cache *cache.Cache
+	Lock  sync.Mutex
```

Comment:
No need to export a Mutex, or?

Reply:
Indeed none of these need to be exported, good catch.